2024-03-03 13:08:19,533 ERROR: Exception on /dictionary/add-term [GET] [in C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\app.py:828]
Traceback (most recent call last):
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\app.py", line 1463, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\app.py", line 872, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\app.py", line 870, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\app.py", line 855, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\app\dictionary\routes.py", line 61, in add_term
    return render_template("dictionary/add_term.html", form=form)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\jinja2\environment.py", line 1301, in render
    self.environment.handle_exception()
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\app\templates\dictionary\add_term.html", line 2, in top-level template code
    {% import "macros_wtf.html" as wtf %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\app\templates\base.html", line 32, in top-level template code
    {% block content %}
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\app\templates\dictionary\add_term.html", line 14, in block 'content'
    {{ render_field(form.definition, value=existing_term.term) }}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\jinja2\environment.py", line 485, in getattr
    return getattr(obj, attribute)
           ^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'existing_term' is undefined
2024-03-03 13:09:22,788 INFO: Decision Support Tool startup [in C:\Users\Ayoub\Desktop\microblog\OAD-CP\app\__init__.py:85]
2024-03-03 13:09:24,379 ERROR: Exception on /dictionary/add-term [GET] [in C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\app.py:828]
Traceback (most recent call last):
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\app.py", line 1463, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\app.py", line 872, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\app.py", line 870, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\app.py", line 855, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\app\dictionary\routes.py", line 61, in add_term
    return render_template("dictionary/add_term.html", form=form)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\jinja2\environment.py", line 1301, in render
    self.environment.handle_exception()
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\app\templates\dictionary\add_term.html", line 2, in top-level template code
    {% import "macros_wtf.html" as wtf %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\app\templates\base.html", line 32, in top-level template code
    {% block content %}
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\app\templates\dictionary\add_term.html", line 24, in block 'content'
    {{ form.definition(rows="13", class="w-full border rounded-md py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline resize-none", value=existing_term.definition) }}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\jinja2\environment.py", line 485, in getattr
    return getattr(obj, attribute)
           ^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'existing_term' is undefined
2024-03-03 13:09:53,115 INFO: Decision Support Tool startup [in C:\Users\Ayoub\Desktop\microblog\OAD-CP\app\__init__.py:85]
2024-03-03 13:09:53,643 ERROR: Exception on /dictionary/add-term [GET] [in C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\app.py:828]
Traceback (most recent call last):
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\app.py", line 1463, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\app.py", line 872, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\app.py", line 870, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\app.py", line 855, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\app\dictionary\routes.py", line 61, in add_term
    return render_template("dictionary/add_term.html", form=form)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\jinja2\environment.py", line 1301, in render
    self.environment.handle_exception()
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\app\templates\dictionary\add_term.html", line 2, in top-level template code
    {% import "macros_wtf.html" as wtf %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\app\templates\base.html", line 32, in top-level template code
    {% block content %}
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\app\templates\dictionary\add_term.html", line 30, in block 'content'
    <a href="{{ redirect(url_for('dictionary.index')) }}" class="bg-gray-500 hover:bg-gray-700 text-white font-bold py-2 px-4 rounded">Cancel</a>
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ayoub\Desktop\microblog\OAD-CP\.venv\Lib\site-packages\jinja2\utils.py", line 83, in from_obj
    if hasattr(obj, "jinja_pass_arg"):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'redirect' is undefined
2024-03-03 13:10:12,944 INFO: Decision Support Tool startup [in C:\Users\Ayoub\Desktop\microblog\OAD-CP\app\__init__.py:85]
2024-03-03 13:56:49,636 INFO: Decision Support Tool startup [in C:\Users\Ayoub\Desktop\microblog\OAD-CP\app\__init__.py:85]
